### PR TITLE
caf:  ble_adv: documentation update

### PR DESCRIPTION
**Pull request by alstrzebonski**
This PR updates documentation, that has to be updated because of this PR: https://github.com/nrfconnect/sdk-nrf/pull/6503.
Fixes misspelling in word "CONFIG" in direct advertising.
This issue caused device to not work properly when direct advertising
option was turned on.

Jira: NCSDK-13058

Signed-off-by: Aleksander Strzebonski <aleksander.strzebonski@nordicsemi.no>

### CODE REVIEW

File: subsys/caf/modules/ble_adv.c [hunk 1]
```diff
@@ -98,9 +98,9 @@ static int settings_set(const char *key, size_t len_rd, settings_read_cb read_cb
 	return 0;
 }
 
-#ifdef COFNIG_CAF_BLE_ADV_DIRECT_ADV
-SETTINGS_STATIC_HANDLER_DEFINE(ble_adv, MODULE_NAME, NULL, settings_set, NULL, NULL);
-#endif /* CONFIG_CAF_BLE_ADV_DIRECT_ADV */
+if (IS_ENABLED(CONFIG_CAF_BLE_ADV_DIRECT_ADV)) {
```
Comment (pdunaj) [inline subsys/caf/modules/ble_adv.c:101]: `if (IS_ENABLED(x))` is to be used inside function body. You can use normal preprocessor code here.

The idea behind `IS_ENABLED` is to always compile excluded code so it is checked for errors. When you use preprocessor the code is excluded before compilation is done. There is so no checks done against the code correctness. `IS_ENABLED` does not really exclude the code from compilation but instead depends on optimizer to remove a dead code when object file is constructed.
This said there is no reason to use `IS_ENABLED` instead of preprocessor outside of function body. The only reason would be the consistency. Still you should use it as `#if IS_ENABLED(x)`. I don't expect anything reasonable will come from placing the code as it was done here. It would be like placing a normal if statement outside of a function.

In summary please use:
```
#if CONFIG_CAF_BLE_ADV_DIRECT_ADV
...
#endif
```
```

**Review by pdunaj — CHANGES_REQUESTED** [review]

**Review by carlescufi — APPROVED** [review]

File: doc/nrf/known_issues.rst [hunk 2]
```diff
@@ -868,6 +868,14 @@ NCSDK-10196: DFU fails for some configurations with the quick session resume fea
 
   **Workaround:** Use the quick session resume feature only for configurations with the cellular network backend.
 
+Common Application Framework (CAF)
+==================================
+
+.. rst-class:: v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0
+
+Directed advertising (:kconfig:`CONFIG_CAF_BLE_ADV_DIRECT_ADV`) does not work properly.
```
Comment (greg-fer) [inline doc/nrf/known_issues.rst:876]: You need to add the related JIRA issue for this known issue. The update requires this part as mandatory, as mentioned in the beginning of the known_issues.rst raw file.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -125,6 +125,8 @@ Common Application Framework (CAF)
     ``MODULE_ID`` macro and :c:func:`module_id_get` function now returns module reference from dedicated section instead of module name.
     The module name can not be obtained from reference object directly, a helper function (:c:func:`module_name_get`) should be used instead.
 
+* Fixed directed advertising.
+
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:129]:
```suggestion
    The module name can not be obtained from reference object directly, a helper function (:c:func:`module_name_get`) should be used instead.
    * Fixed the known issue related to directed advertising in CAF.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -142,6 +142,7 @@ Common Application Framework (CAF)
     The array holding module reference objects is explicitly defined in linker script to avoid creating an orphan section.
     ``MODULE_ID`` macro and :c:func:`module_id_get` function now returns module reference from dedicated section instead of module name.
     The module name can not be obtained from reference object directly, a helper function (:c:func:`module_name_get`) should be used instead.
+  * Fixed the known issue related to directed advertising in CAF.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:145]:
```suggestion
 * Fixed the NCSDK-13058 known issue related to directed advertising in CAF.
```

File: doc/nrf/known_issues.rst [hunk 5]
```diff
@@ -868,6 +868,16 @@ NCSDK-10196: DFU fails for some configurations with the quick session resume fea
...
 
+Common Application Framework (CAF)
+==================================
+
+.. rst-class:: v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0
+
+NCSDK-13058: Directed advertising (:kconfig:`CONFIG_CAF_BLE_ADV_DIRECT_ADV`) does not work properly.
+  Device is unable to connect to host, when it uses directed advertising.
+
+  **Workaround** Do not use directed advertising.
```
Comment (greg-fer) [inline doc/nrf/known_issues.rst:879]:
```suggestion
  **Workaround** Do not use directed advertising in CAF.
```

File: doc/nrf/known_issues.rst [hunk 6]
```diff
@@ -868,6 +868,16 @@ NCSDK-10196: DFU fails for some configurations with the quick session resume fea
 
   **Workaround:** Use the quick session resume feature only for configurations with the cellular network backend.
 
+Common Application Framework (CAF)
+==================================
+
+.. rst-class:: v1-8-0 v1-7-1 v1-7-0 v1-6-1 v1-6-0
+
+NCSDK-13058: Directed advertising (:kconfig:`CONFIG_CAF_BLE_ADV_DIRECT_ADV`) does not work properly.
+  Device is unable to connect to host, when it uses directed advertising.
```
Comment (greg-fer) [inline doc/nrf/known_issues.rst:877]:
```suggestion
NCSDK-13058: Directed advertising does not work
  The directed advertising feature enabled with the :kconfig:`CONFIG_CAF_BLE_ADV_DIRECT_ADV` option does not work as intended.
  The device is unable to connect to the host when it uses directed advertising.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/known_issues.rst (hunk 6 above)
Comment (MarekPieta) [reply doc/nrf/known_issues.rst:877]: Actually this is partially true. The problem affects only peers with RPA that do not support resolution of directed advertisements (we can check this by reading the Central Address Resolution Characteristic). Also the problem is related only to loading settings, so it would not be visible before reboot. To simplify implementation, we considered using direct advertising only for some time and then switch to indirect advertising, See `NCSDK-13128` for details.

**Review by pdunaj — APPROVED** [review]

File: doc/nrf/known_issues.rst (hunk 6 above)
Comment (MarekPieta) [reply doc/nrf/known_issues.rst:877]: Maybe we should state that `direct advertising towards privacy enabled peers may not work` to avoid explaining all of the details?

Also, please add `cherry-pick commit_hash` as a workaround. I think that is way more useful information - fix can be easily cherry-picked (then fix needs to go in a separate PR and you can refer to commit hash from main).

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: doc/nrf/known_issues.rst (hunk 6 above)
Comment (greg-fer) [reply doc/nrf/known_issues.rst:877]: Sounds reasonable.

Comment (greg-fer) [reply doc/nrf/known_issues.rst:877]: @MarekPieta , what are `privacy-enabled peers`?

**Comment by alstrzebonski** [comment]
I've changed this PR to documentation PR. It can only be merged after PR: https://github.com/nrfconnect/sdk-nrf/pull/6503, because I need to mention commit hash in documentation.

File: doc/nrf/known_issues.rst (hunk 6 above)
Comment (MarekPieta) [reply doc/nrf/known_issues.rst]: The peers that enable BLE privacy and use RPA (Resolvable Private Address),

Comment (greg-fer) [reply doc/nrf/known_issues.rst]: @MarekPieta , @alstrzebonski , then I'd suggest:
```suggestion
NCSDK-13058: Directed advertising does not work
  The directed advertising feature enabled with the :kconfig:`CONFIG_CAF_BLE_ADV_DIRECT_ADV` option does not work as intended.
  Using directed advertising towards peers that enable privacy may result in connection establishing problems
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
@alstrzebonski , check my reply to comment from @MarekPieta for the latest suggestion.

**Review by greg-fer — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]
